src/lib.rs

```diff
@@ -116,6 +111,18 @@ fn map(file: &File) -> Result<MmapMut> {
     }
 }
 
+/// Trait for when `bincode`'s overhead is unacceptable and the developer would prefer to
+/// directly write the message's contents into the mmap buffer without an additional copy and/or serialization.
+///
+/// `msg_len` is the encoded size of the message in bytes. 
+/// `write_to_shm` allows for directly copying contents of the implementing struct into the 
+/// message's space in the buffer. This is incredibly unsafe and fully up to the developer
+/// to account for padding, alignment, proper offsetting etc. or you will cause mayhem.
+pub trait ShmWriter {
+    fn msg_len(&self) -> u32;
+    fn write_to_shm(&self, shm_ptr: *mut u8, msg_len: u32) -> Result<()>;
```

Comment:
What's the advantage of using a raw pointer here rather than a `&mut [u8]` slice?  That would make the API safe, but still allow the implementer to convert the slice to a raw pointer if they want to do something unusual (in which case soundness is their responsibility).

Reply:
I don't necessarily think there is an advantage for using a raw pointer, probably more so a result of some bad habits that have been picked up over time haha. Will update to use `&mut [u8]` as it does give more flexibility